client/src/redux/sagas/datafiles.sagas.js

```diff
@@ -596,3 +597,178 @@ export function* trashFile(system, path, id) {
     });
   }
 }
+
+export const getLatestApp = async name => {
+  const res = await fetchUtil({
+    url: '/api/workspace/apps',
+    params: { publicOnly: true }
+  });
+  const apps = res.response;
+
+  const latestApp = apps
+    .filter(app => app.id.includes(name))
+    .reduce(
+      (latest, app) => {
+        if (app.version > latest.version) {
+          return app;
+        }
+        if (app.version < latest.version) {
+          return latest;
+        }
+        // Same version of app
+        if (app.revision >= latest.revision) {
+          return app;
+        }
+        return latest;
+      },
+      { revision: null, version: null }
+    );
+  return latestApp.id;
+};
+
+const getExtractParams = (file, latestExtract) => {
+  const inputFile = `agave://${file.system}${file.path}`;
+  const archivePath = `agave://${file.system}${file.path.substring(
+    0,
+    file.path.lastIndexOf('/') + 1
+  )}`;
+  return JSON.stringify({
+    allocation: 'FORK',
+    appId: latestExtract,
+    archive: true,
+    archivePath,
+    inputs: {
+      inputFile
+    },
+    maxRunTime: '02:00:00',
+    name: 'Extracting Zip File',
```

Comment:
should this be more generic?  like maybe "Extracting Compressed File" or "Extracting File"